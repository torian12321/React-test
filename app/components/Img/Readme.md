Basic:
```
<Img />
```

With alt text:
```
<Img alt='Some text'/>
```