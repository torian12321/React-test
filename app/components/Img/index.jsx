var
React     = require('react'),
PropTypes = require('prop-types');

const Img = props => (
    <img 
        src      ={props.src}
        className={props.className}
        alt      ={props.alt || null}
        role     ={props.alt ? null : 'presentation'}
    />
);

Img.propTypes = {
    src      : PropTypes.string.isRequired,
    alt      : PropTypes.string,
    className: PropTypes.string
};
Img.defaultProps = {
    src      : './src/img/default/def.png',
    className: null
};

module.exports = Img;
