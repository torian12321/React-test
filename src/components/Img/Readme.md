We require the use of src and alt, only enforced by react in dev mode

#### Basic
Use this just for wireframing
```
<Img alt="desc" />
```

#### Custom src
Use a local image or from a url
```
const imgExample = require ('./example.jpg');

<div>
  <Img alt="img 1" src='http://placeimg.com/200/150/nature' />
  <Img alt="img 2" src={imgExample} />
</div>
```
