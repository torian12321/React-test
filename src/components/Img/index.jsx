import React from 'react';
import PropTypes from 'prop-types';
import ImgDef from '../../img/default/img.png';

const Img = (props) => {
    const { src, alt, ...other } = props;

    return (
        <img
            {...other}
            src ={src}
            alt ={alt}
            role={alt ? null : 'presentation'}
        />
    );
};

Img.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired
};
Img.defaultProps = {
    //src: 'http://placeimg.com/200/150/nature'
    src: ImgDef
};

module.exports = Img;
