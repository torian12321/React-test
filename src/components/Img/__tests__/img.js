import React    from 'react';
import renderer from 'react-test-renderer';
import Img      from '../index';

jest.unmock('../index.jsx');

test('Should render Img', () => {
  const component = renderer.create(
    <Img src='img.png' alt='image description' />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});