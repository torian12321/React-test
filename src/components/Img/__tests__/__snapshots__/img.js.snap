// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Img 1`] = `
<img
  alt="image description"
  role={null}
  src="img.png"
/>
`;
